File: frontend/lib/shared/result.dart
// Result abstract and Ok and Err classes

abstract class Result<T> {
  const Result();
}

class Ok<T> extends Result<T> {
  final T value;

  const Ok(this.value);
}

class Err<T> extends Result<T> {
  final String error;

  const Err(this.error);
}

File: backend/src/shared/models/result.ts
// Result interface and Ok and Error classes implementation

import { error } from "console";

export interface Result<T> {
  isSuccess: boolean;
  isError: boolean;
  value?: T;
  error?: Error;
}

export class Ok<T> implements Result<T> {
  isSuccess = true;
  isError = false;
  constructor(public value: T) {}
}

export class Err implements Result<never> {
  isSuccess = false;
  isError = true;
  constructor(public error: Error) {}
}